File: platform/app/schemas/skill.py
from __future__ import annotations

from datetime import datetime
from typing import Dict, List, Optional

from pydantic import BaseModel


class SkillCreateRequest(BaseModel):
    name: str
    display_name: str
    description: Optional[str] = None
    layer: str = "L1"
    tags: Optional[List[str]] = None
    applicable_roles: Optional[List[str]] = None
    content: Optional[str] = None
    git_path: Optional[str] = None


class SkillUpdateRequest(BaseModel):
    display_name: Optional[str] = None
    description: Optional[str] = None
    layer: Optional[str] = None
    tags: Optional[List[str]] = None
    applicable_roles: Optional[List[str]] = None
    content: Optional[str] = None
    git_path: Optional[str] = None
    version: Optional[str] = None


class SkillDetailResponse(BaseModel):
    id: str
    name: str
    display_name: str
    description: Optional[str] = None
    layer: str
    tags: Optional[List[str]] = None
    applicable_roles: Optional[List[str]] = None
    status: str
    version: str
    content: Optional[str] = None
    git_path: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SkillListResponse(BaseModel):
    items: List[SkillDetailResponse]
    total: int
    page: int
    page_size: int


class SkillStatsResponse(BaseModel):
    total: int
    by_layer: Dict[str, int]
    by_status: Dict[str, int]

File: web/src/types/skill.ts
export interface Skill {
  id: string;
  name: string;
  display_name: string;
  description: string | null;
  layer: string;
  tags: string[] | null;
  applicable_roles: string[] | null;
  version: string;
  status: string;
  content: string | null;
  git_path: string | null;
  created_at: string;
  updated_at: string;
}

export interface SkillVersion {
  version: string;
  created_at: string;
  current: boolean;
  change_summary?: string;
}

export interface SkillCreateRequest {
  name: string;
  display_name: string;
  description?: string;
  layer?: string;
  tags?: string[];
  applicable_roles?: string[];
  content?: string;
  git_path?: string;
}

export interface SkillUpdateRequest {
  display_name?: string;
  description?: string;
  layer?: string;
  tags?: string[] | null;
  applicable_roles?: string[];
  content?: string | null;
  git_path?: string | null;
  version?: string;
}

export interface SkillListResponse {
  items: Skill[];
  total: number;
  page: number;
  page_size: number;
}

export interface SkillVersionListResponse {
  name: string;
  versions: SkillVersion[];
}

export interface SkillStatsResponse {
  total: number;
  by_layer: Record<string, number>;
  by_status: Record<string, number>;
}

export interface SkillSyncResponse {
  synced: number;
  created: number;
  updated: number;
  details: Record<string, string>;
}
